src/sysbench.c

```diff
@@ -1444,6 +1445,17 @@ int main(int argc, char *argv[])
   sb_globals.argv = malloc(argc * sizeof(char *));
   memcpy(sb_globals.argv, argv, argc * sizeof(char *));
 
+  /* Get the environment's size and copy it to sb_globals */
+  int env_size = 0;
+  while (env[env_size] != NULL) {
+    env_size++;
+  }
+  sb_globals.env = malloc(env_size * sizeof(char *));
+  if (sb_globals.env == NULL) {
+      return EXIT_FAILURE;
+  }
+  memcpy(sb_globals.env, env, env_size * sizeof(char *));
+
```

Comment:
This is not right. You forgot that the `env` array must always be `NULL`-terminated, otherwise users of the array will not know how long it is. You need to allocate and copy `env_size + 1` to keep that last `NULL` entry.

Reply:
Right, I've added a `NULL` row at the very end of the `char **` and did the same thing in `sb_tpch.c` where I was also creating a `char **`